ansible/roles/matrix/tasks/main.yml

```diff
@@ -1,29 +1,69 @@
 ---
-- include: tasks/docker.yml
+- name: Install required packages
+  apt: name={{ item }} state=latest
+  with_items:
+    - build-essential
+    - python2.7-dev
+    - libffi-dev 
+    - python-pip
+    - python-setuptools
+    - sqlite3 
+    - libssl-dev
+    - libjpeg-dev
+    - libxslt1-dev
 
-- name: Create supervisor config
-  copy:
-    src: supervisor/matrix.conf
-    dest: /etc/supervisor/conf.d/matrix.conf
-
-- name: Create nginx config
-  template:
-    src: nginx/matrix.conf
-    dest: /etc/nginx/sites-enabled/matrix.conf
-
-- name: Create the libexec folder
+- name: Create the matrix folder
   file:
-    path: /opt/matrix/libexec
+    path: /opt/matrix
     state: directory
     mode: 0755
 
-- name: Create shell configuration file
+- name: Subscribe to hooks
+  copy:
+    src: hooks/user
+    dest: /opt/common/hooks/{{ item }}.d/24-matrix.sh
+    mode: 0755
+  with_items:
+    - user-created
+
+- name: Copy the requirements file
+  copy:
+    src: requirements.txt
+    dest: /opt/matrix/requirements.txt
+
+- name: Install synapse homeserver
+  pip:
+    requirements: /opt/matrix/requirements.txt
+    virtualenv: /opt/matrix/synapse
+    virtualenv_python: python2.7
+
+- name: Run the default configuration
+  command: /opt/matrix/synapse/bin/python -m synapse.app.homeserver --server-name matrix.{{ liquid_domain }} --config-path homeserver.yaml --generate-config --report-stats=no
+  args:
+    chdir: /opt/matrix/synapse
+    creates: /opt/matrix/synapse/homeserver.yaml
+
+- name: Overwrite the configuration file
```

Comment:
Why are we overwriting the matrix config file?

Reply:
Because we have to change the configs. See the previous script that generates a bunch of other config files as well.

Reply:
Yeah, I assumed as much. But what's different in our file?

Reply:
Hostname, protocol, ports (because we go through nginx)

Reply:
Then at least let's trim out the comments in our config file template.